include/cgimap/xml_writer.hpp

```diff
@@ -38,24 +42,41 @@ class xml_writer : public output_writer {
   ~xml_writer() noexcept override;
 
   // begin a new element with the given name
-  void start(const std::string &name);
+  void start(const char *name);
+  inline void start(const std::string &name) { start(name.c_str()); }
 
   // write an attribute of the form name="value" to the current element
-  void attribute(const std::string &name, const std::string &value);
+  void attribute(const char *name, const std::string &value);
 
   // write a mysql string, which can be null
-  void attribute(const std::string &name, const char *value);
+  void attribute(const char *name, const char *value);
 
   // overloaded versions of writeAttribute for convenience
-  void attribute(const std::string &name, double value);
-  void attribute(const std::string &name, int32_t value);
-  void attribute(const std::string &name, int64_t value);
-  void attribute(const std::string &name, uint32_t value);
-  void attribute(const std::string &name, uint64_t value);
-  void attribute(const std::string &name, bool value);
+  void attribute(const char *name, double value);
+  void attribute(const char *name, bool value);
```

Comment:
Since `std::is_integral_v<bool> == true`, I would somehow expect full template specialization for "bool" instead of an overloaded version: https://cpp.godbolt.org/z/M1GTev3jv